pkg/operator/osd/spec.go

```diff
@@ -24,8 +24,9 @@ import (
 
 // StorageSpec CRD settings
 type StorageSpec struct {
-	Nodes       []Node `json:"nodes,omitempty"`
-	UseAllNodes bool   `json:"useAllNodes,omitempty"`
+	Nodes       []Node      `json:"nodes,omitempty"`
+	UseAllNodes bool        `json:"useAllNodes,omitempty"`
+	Directories []Directory `json:"directories,omitempty"`
```

Comment:
instead of duplicating this `Directories` field in the `StorageSpec` type, I would look into refactoring it into the `Selection` type: https://github.com/rook/rook/blob/master/pkg/operator/osd/spec.go#L54

That way, both `StorageSpec` and `Node` get it for free and there's no duplication.